### PR TITLE
cl-adm: add version flag to CLI

### PR DESCRIPTION
**Pull request by kfirtoledo**
Add a version flag to create and deploy peers to be able to deploy different versions of ClusterLink (not just the latest).

### CODE REVIEW

File: cmd/cl-adm/cmd/create/create_peer.go [hunk 1]
```diff
@@ -58,6 +60,7 @@ func (o *PeerOptions) AddFlags(fs *pflag.FlagSet) {
 		"The log level. One of fatal, error, warn, info, debug.")
 	fs.StringVar(&o.ContainerRegistry, "container-registry", config.DefaultRegistry,
 		"The container registry to pull the project images. If empty will use local registry.")
+	fs.StringVar(&o.Version, "version", "latest", "The version of the project images.")
```
Comment (elevran) [inline cmd/cl-adm/cmd/create/create_peer.go:63]: nit: I think it's called an image `tag` and not a `version`. Using `release tag` might also be ok

File: design-proposals/project-deploymnet.md [hunk 2]
```diff
@@ -121,7 +121,7 @@ The ClusterLink CRD includes the following fields:
     ||Port| Port represents the port number of the external service |443 for all types,except for NodePort, where the port number will be allocated by Kubernetes | 
     |logLevel| |Log level severity for all the components (controlplane and dataplane)| "info"|
     |containerRegistry| |The container registry to pull the project images when the images is not present locally | ghcr.io/clusterlink-net|
-    |imageTag| |The project images version | latest|
+    |version| |The project images version | latest|
```
Comment (elevran) [inline design-proposals/project-deploymnet.md:124]: nit: I prefer the use of `tag`

File: pkg/operator/controller/instance_controller_test.go [hunk 3]
```diff
@@ -235,7 +235,7 @@ func TestClusterLinkController(t *testing.T) {
 		goReplicas := 2
 		loglevel := "debug"
 		containerRegistry := "quay.com"
-		imageTag := "v1.0.1"
+		version := "v1.0.1"
```
Comment (elevran) [inline pkg/operator/controller/instance_controller_test.go:238]: not `v0.0.1` 😜 ?

**Review by elevran — COMMENTED** [review]
considerable-sized change which mostly renames `tag` to `version`. I'm also not sure `version` is more appropriate than `tag` when talking about images

**Review by praveingk — APPROVED** [review]
LGTM

**Pull request merged**